internal/newrelic/client.go

```diff
@@ -142,17 +146,100 @@ func nrqlQueryExpectedValueTest(nrc *nrClient, query string, expectedResults []s
 	a, _ := nrc.client.Query(nrc.accountID, query)
 
 	if len(expectedResults) != len(a.Results) {
-		return fmt.Errorf("%w: %s - expected %d got %d", ErrResultNumber, query, len(expectedResults), len(a.Results))
+		return fmt.Errorf("%w: %s\n - expected %d got %d", ErrResultNumber, query, len(expectedResults), len(a.Results))
 	}
 	for i, expectedResult := range expectedResults {
-		stringResult := fmt.Sprintf("%v", a.Results[i][expectedResult.Key])
-		if stringResult != expectedResult.Value {
-			return fmt.Errorf("%w: %s - expected for key '%s': '%s' got '%s'", ErrNotExpectedResult, query, expectedResult.Key, expectedResult.Value, stringResult)
+		actualResult := a.Results[i][expectedResult.Key]
+		comparisonErr := compareResults(actualResult, expectedResult)
+		if comparisonErr != nil {
+			return fmt.Errorf("%w: %s\n - for key '%s': %s", ErrNotExpectedResult, query, expectedResult.Key, comparisonErr.Error())
 		}
 	}
 	return nil
 }
 
+func compareResults(actualResult any, expectedResult spec.TestNRQLExpectedResult) error {
+	expectedExactResult := expectedResult.Value
+	if expectedExactResult != nil {
+		// We are checking for an exact value
+		expectedExactResult = preprocessResult(expectedExactResult)
+		actualResult = preprocessResult(actualResult)
+
+		if expectedExactResult == actualResult {
+			return nil
+		}
+		return fmt.Errorf("%w - expected: '%s', got '%s'", ErrAssertionFailure, expectedExactResult, actualResult)
+	}
+
+	// We are checking for a bounded value
+	return checkBounds(actualResult, expectedResult.LowerBoundedValue, expectedResult.UpperBoundedValue)
```

Comment:
should add a short circuit error that checks either before line 175 here, or first thing in the function `checkBounds` to make sure that at least either `LowerBoundedValue` or `UpperBoundedValue` are populated.

Reply:
We actually do ensure that one or both are populated. So, this function is never called if that is the case.

Reply:
are you referring to the check that is happening at the end of the `checkBounds` function?

Reply:
oh wait, you said this function is never called if thats the case, maybe i haven't gotten to that part of the pr yet